### DISCUSSION TITLE
Just saying thank you

### DISCUSSION THREAD

**Discussion by The-Paavo** [lkrms/pretty-php#38] [General]
I have no complaints, but I just wanted to thank you. This is exactly what I need. Well done, sir! 

A couple of quick questions, though.

- First one is a selfish question. How about making `pretty-php.formatting.psr12: true` as the default value in your VS Code extension? Reason for that: Because it is the only option I changed. 🤣 
- Is `.prettyphp` file documented somewhere?

**Comment by lkrms** [comment]
Thanks for the encouraging feedback!

I hear you re: the PSR-12 setting and would like to enable it by default, but there are a couple of [pending compliance tasks](https://github.com/lkrms/pretty-php/issues/4) that, when complete, might come as a surprise to existing users, so my current plan is to hold off on that until those changes are stable. 

`.prettyphp` isn't documented comprehensively but there is some info in `pretty-php --help` (which I'm planning to include in the README soon). Values correspond to command-line option names in camelCase.

> **CONFIGURATION**
>
> `pretty-php` looks for a JSON configuration file named _.prettyphp_ or _prettyphp.json_ in the same directory as each input file, then in each of its parent directories. It stops looking when it finds a configuration file, a _.git_ directory, a _.hg_ directory, or the root of the filesystem, whichever comes first.
>
> If a directory contains more than one configuration file, `pretty-php` reports an error and exits without formatting anything.
>
> For input files with an applicable configuration file, command-line formatting options other than `-N/--ignore-newlines` are replaced with settings from the configuration file.
>
> The `--print-config` option can be used to generate a configuration file, for example:
>
>     $ pretty-php -P -S -M --print-config src tests bootstrap.php
>     {
>         "src": [
>             "src",
>             "tests",
>             "bootstrap.php"
>         ],
>         "includeIfPhp": true,
>         "noSimplifyStrings": true,
>         "noSortImports": true
>     }
>
> The optional _src_ array specifies files and directories to format. If `pretty-php` is started with no _PATH_ arguments in a directory where _src_ is configured, or the directory is passed to `pretty-php` for formatting, paths in _src_ are formatted. It is ignored otherwise.

**Comment by The-Paavo** [comment]
> .prettyphp isn't documented comprehensively but there is some info in pretty-php --help (which I'm planning to include in the README soon). Values correspond to command-line option names in camelCase.

That's a good plan. 👍 

Some (if not all) VS Code extension users aren't directly executing `pretty-php` and may be unaware that `pretty-php.phar` even exists. We simply use the bundled version via Visual Studio Code.

**Comment by lkrms** [comment]
Apologies for not following up sooner, but I'm hoping progress made over the last few months has addressed your concerns about configuring `pretty-php`:

- A JSON Schema for `.prettyphp`/`prettyphp.json` files is now bundled with the VS Code extension, which means they can be edited with IntelliSense hints.
- Config file documentation has been added to the `README.md` files of `pretty-php` and the VS Code extension.
- The VS Code extension now has a "Create .prettyphp or prettyphp.json" command and context menu item that applies your current VS Code settings to a new or existing config file.

Thanks again for your feedback 😊
